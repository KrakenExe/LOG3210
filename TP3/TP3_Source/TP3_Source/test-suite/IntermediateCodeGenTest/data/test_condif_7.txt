Declaration
num a;
bool b;
num c;
bool d;
EndDeclaration

if(b && d) {
  a = 42;
  if(d || false) {
    a = a + 100;
  }
  else {
    a = a - 100;
  }
  c = 1 + 2;
}
else {
  a = 3210;
  c = a * 4;
}

