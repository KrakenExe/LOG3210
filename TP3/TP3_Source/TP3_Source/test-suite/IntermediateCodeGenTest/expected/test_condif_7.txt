if b == 1 goto _L3
goto _L2
_L3
if d == 1 goto _L1
goto _L2
_L1
a = 42
_L4
if d == 1 goto _L6
goto _L8
_L8
goto _L7
_L6
_t0 = a + 100
a = _t0
goto _L5
_L7
_t1 = a - 100
a = _t1
_L5
_t2 = 1 + 2
c = _t2
goto _L0
_L2
a = 3210
_L9
_t3 = a * 4
c = _t3
_L0
